src/lib/textToEdit.ts

```diff
@@ -0,0 +1,351 @@
+        } else {
+          prompts.push({
+            prompt: `This selection is for a segment (line, xLine, angledLine etc) that has been swept (a general-sweep, either an extrusion, revolve, sweep or loft).
+Because it now refers to an edge the way to refer to this edge is to add a tag to the segment, and then use that tag directly.
+i.e. \`fillet({ radius = someInteger, tags = [newTag] }, %)\` will work in the case of filleting this edge
+See later source ranges for more context. about the sweep`,
+            range: convertAppRangeToApiRange(selection.codeRef.range, code),
+          })
+          let path = getArtifactOfTypes(
+            { key: artifact.pathId, types: ['path'] },
+            artifactGraph
+          )
+          if (!err(path)) {
+            const sweep = getArtifactOfTypes(
+              { key: path.sweepId, types: ['sweep'] },
+              artifactGraph
+            )
+            if (!err(sweep)) {
+              prompts.push({
+                prompt: `This is the sweep's source range from the user's main selection of the edge.`,
+                range: convertAppRangeToApiRange(sweep.codeRef.range, code),
+              })
+            }
+          }
+        }
+      }
+      return prompts
+    })
+  console.log('prompts')
+  ranges.map((a) => console.log(a.prompt))
+  const body: Models['TextToCadIterationBody_type'] = {
+    original_source_code: code,
+    prompt,
+    source_ranges: ranges,
+  }
+  const url = VITE_KC_API_BASE_URL + '/ml/text-to-cad/iteration'
+  const data: Models['TextToCadIteration_type'] | Error =
+    await crossPlatformFetch(
+      url,
+      {
+        method: 'POST',
+        body: JSON.stringify(body),
+      },
+      token
+    )
+
+  // Make sure we have an id.
+  if (data instanceof Error) {
+    return data
+  }
+
+  if (!data.id) {
+    return new Error('No id returned from Text-to-CAD API')
+  }
+  console.log('data', data)
+  return data
+}
+
+export async function getPromptToEditResult(
+  id: string,
+  token?: string
+): Promise<Models['TextToCadIteration_type'] | Error> {
+  const url = VITE_KC_API_BASE_URL + '/async/operations/' + id
+  const data: Models['TextToCadIteration_type'] | Error =
+    await crossPlatformFetch(
+      url,
+      {
+        method: 'GET',
+      },
+      token
+    )
+
+  return data
+}
+
+export async function doPromptEdit({
+  prompt,
+  selections,
+  code,
+  token,
+  artifactGraph,
+}: {
+  prompt: string
+  selections: Selections
+  code: string
+  token?: string
+  artifactGraph: ArtifactGraph
+}): Promise<Models['TextToCadIteration_type'] | Error> {
+  const toastId = toast.loading('Submitting to Edit AI...')
+  const submitResult = await submitPromptToEditToQueue({
+    prompt,
+    selections,
+    code,
+    token,
+    artifactGraph,
+  })
+  if (err(submitResult)) return submitResult
+
+  const textToCadComplete = new Promise<Models['TextToCadIteration_type']>(
+    (resolve, reject) => {
+      ;(async () => {
```

Comment:
This async func and the new Promise((r) => setTimeout(r, ..))) are the only things that make me kinda go 'wat'

Reply:
the `;(async () => {` is there only because `new Promise(() => {})` does not like getting an async function, which doesn't really compute with me, and then `new Promise(() => {})` is a throttling mechanism so that we leave a 200ms break between one response from the API before polling again, Frank did this differently in `src/components/ToastTextToCad.tsx` using a setInterval instead, that smells to me because it feels like a race potentially you could fire of a request before the last one came back.

Reply:
Open to better ways to polling though.